src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+        }
+        return null;
+    }
+
+    /**
+     * Define which discount fare should be applied based on the fare type. If the ride type is unknown the discount
+     * fare can not be applied, use the default fare.
+     */
+    private float getDiscountedFare(Fare.FareType fareType, RideType rideType, float defaultFare) {
+        if (rideType == null) {
+            return defaultFare;
+        }
+        switch (fareType) {
+            case youth:
+            case orcaYouth:
+                return getYouthFare(rideType);
+            case orcaLift:
+                return getLiftFare(rideType, defaultFare);
+            case orcaSenior:
+            case senior:
+                return getSeniorFare(fareType, rideType, defaultFare);
+            case orcaRegular:
+                return getRegularFare(rideType, defaultFare);
+            case regular:
+            default:
+                return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca regular discount fares. If the ride type can not be matched the default fare is used.
+     */
+    private float getRegularFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case KC_WATER_TAXI_VASHON_ISLAND: return 5.75f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 5.00f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Apply Orca lift discount fares based on the ride type.
+     */
+    private float getLiftFare(RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case PIERCE_COUNTY_TRANSIT: return defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 1.00f;
+            // KCM, Sound Transit, Everett, Seattle Streetcar.
+            default: return 1.50f;
+        }
+    }
+
+    /**
+     * Apply Orca senior discount fares based on the fare and ride types.
+     */
+    private float getSeniorFare(Fare.FareType fareType, RideType rideType, float defaultFare) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case EVERETT_TRANSIT:
+                return fareType.equals(Fare.FareType.orcaSenior) ? 0.50f : defaultFare;
+            case PIERCE_COUNTY_TRANSIT:
+            case SEATTLE_STREET_CAR:
+            case KITSAP_TRANSIT:
+                // Pierce, Seattle Streetcar, and Kitsap only provide discounted senior fare for orca.
+                return fareType.equals(Fare.FareType.orcaSenior) ? 1.00f : defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 3.00f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 2.50f;
+            // KC Metro, Sound Transit
+            default: return 1.00f;
```

Comment:
If we continue using switch statements like this, the default should always return the default fare in case there isn't an exact transit agency to match. This will future-proof things in case additional agencies get added in the future.